crates/sui-indexer/src/handlers/pruner.rs

```diff
@@ -38,61 +41,52 @@ impl Pruner {
     }
 
     pub async fn start(&self, cancel: CancellationToken) -> IndexerResult<()> {
-        let mut last_seen_max_epoch = 0;
+        // Spawn a separate task to continuously update the watermarks for the reader. We can't
+        // reliably update watermarks while we're in the middle of a pruning operation. This is
+        // because the pruning operation itself may take a considerable amount of time, during which
+        // the system state could change significantly. Pruner prunes on the main thread to limit
+        // concurrency.
+        let store_clone = self.store.clone();
+        let epochs_to_keep = self.epochs_to_keep;
+        let cancel_clone = cancel.clone();
+        spawn_monitored_task!(update_watermarks_lower_bounds_task(
+            store_clone,
+            epochs_to_keep,
+            cancel_clone
+        ));
+
+        // Similarly, handle epoch-partitioned tables in a separate task, so that the main thread
+        // can focus on the slowest pruning operation.
```

Comment:
I'm playing catch up here: Why is it beneficial for the main thread to handle the slowest pruning operation? (Related question, why is it beneficial to "limit concurrency"?)